src/games/strategy/engine/framework/startup/mc/GameSelectorModel.java

```diff
@@ -206,32 +195,28 @@ public void load(final File file, final Component ui) {
   }
 
   public boolean isSavedGame() {
-    return !m_fileName.endsWith(".xml");
+    return m_fileName == null || !m_fileName.endsWith(".xml");
   }
 
   private void error(final String message, final Component ui) {
     JOptionPane.showMessageDialog(JOptionPane.getFrameForComponent(ui), message, "Could not load Game", JOptionPane.ERROR_MESSAGE);
   }
 
-  public synchronized GameData getGameData() {
```

Comment:
- The test coming out will be a unit test for gameSelector. As part of that I've have gone through all the public methods and the callers of those methods. There does not look to be any race conditions we need to worry about. 
- Secondly, the synchronization is not correct and looks to be pretty arbitrary and thrown down without much of an understanding of synchronization. For example, synchronizing around "this" for setting or returning "m_canSelect" is needless. Synchronization is not needed around a single atomic operation (and returning the value of a local private boolean variable is atomic)